pandas/core/indexes/multi.py

```diff
@@ -1760,9 +1799,10 @@ def __setstate__(self, state):
 
         self._set_levels([Index(x) for x in levels], validate=False)
         self._set_codes(codes)
+        new_codes = self._verify_integrity()
+        self._set_codes(new_codes)
```

Comment:
don't we end up calling verify_integrity twice? here (and in _set_codes)?

Reply:
Now that the set_codes are called with default verify_integrity=False, verification would not be called twice